src/aave-v3/SupplyVault.sol

```diff
@@ -180,16 +167,20 @@ contract SupplyVault is ISupplyVault, SupplyVaultBase {
             address(this),
             _rewardToken
         );
+
         UserRewardsData memory rewards = userRewards[_rewardToken][_user];
+        uint256 supply = totalSupply();
 
-        return
-            rewards.unclaimed +
-            balanceOf(_user).mulDivDown(
-                (rewardsIndex[_rewardToken] +
-                    claimableRewards.mulDivDown(SCALE, totalSupply()) -
-                    rewards.index),
-                SCALE
-            );
+        if (supply > 0)
+            return
+                rewards.unclaimed +
+                balanceOf(_user).mulDivDown(
+                    (rewardsIndex[_rewardToken] +
+                        claimableRewards.mulDivDown(SCALE, supply) -
+                        rewards.index),
+                    SCALE
+                );
+        return rewards.unclaimed;
```

Comment:
```suggestion
        if (supply == 0) return rewards.unclaimed;
        return
            rewards.unclaimed +
            balanceOf(_user).mulDivDown(
                (rewardsIndex[_rewardToken] +
                    claimableRewards.mulDivDown(SCALE, supply) -
                    rewards.index),
                SCALE
             );
```
Small and optional style change

Reply:
I won't change it since it's refactored in another PR....